packages/@azure/servicebus/data-plane/lib/client.ts

```diff
@@ -8,20 +8,25 @@ import { AmqpError, generate_uuid } from "rhea-promise";
 import { throwErrorIfConnectionClosed } from "./util/utils";
 
 /**
- * Describes the base class for a client.
+ * Describes the abstract base class for QueueClient, TopicClient and SubscriptionClient
  * @abstract
  * @class Client
  */
 export abstract class Client {
   /**
-   * @property {string} name The name of the entity (queue, topic, subscription, etc.)
+   * @property {string} The entitypath for the Service Bus entity for which this client is created.
+   * @readonly
    */
-  name: string;
+  get entityPath(): string {
+    return this._entityPath;
+  }
   /**
-   * @property {string} id A unique identifier for the client. It is usually a combination of
-   * the name and a Guid.
+   * @property {string} A unique identifier for the client.
+   * @readonly
    */
-  id: string;
+  get id(): string {
```

Comment:
I'm not sure moving to getters is necessary. Can we just use TypeScript's readonly modifier?

Reply:
perfect, somehow that totally skipped my mind

Reply:
done